### ISSUE TITLE
visual studio 2019.3 timeouts while build F# projects in parallel

### ISSUE THREAD

**Issue by jmecosta** [dotnet/fsharp#7683]
Hi,

how can i investigate a timeout while building several projects in parallel... basically msbuild processes are still running! but waiting it seems for fsc to do something...  ive notice some crashes of fsc.exe also locally related wtih "dll build from previous version" but not sure that is related.

it happens only with /m, with /m:1 then it doesnt lock..

this started after using sdk based projects mixed with c++ old projects and with central https://github.com/microsoft/MSBuildSdks/tree/master/src/CentralPackageVersions 

Timeouts started only after forcing those fsharp projects to use that central package... just would like to have a way to debug this.

**Comment by jmecosta** [comment]
heres is a example:

Fsc (running for 10m:27s)
[18:38:45][Fsc] C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\Common7\IDE\CommonExtensions\Microsoft\FSharp\fsc.exe -o:E:\prod\ts-build\MSBuild\..\ObjDrop\Work\Debug\x64\FarmiCaseHelpers.Test\FarmiCaseHelpers.Test.dll
-g
--debug:full
--noframework
--define:TRACE
--define:DEBUG
--define:NETFRAMEWORK
--define:NET472
--optimize-
--tailcalls-
-r:C:\Users\buildmaster\.nuget\packages\atlassian.sdk\10.6.1\lib\net452\Atlassian.Jira.dll
-r:C:\Users\buildmaster\.nuget\packages\diffplex\1.4.4\lib\net40\DiffPlex.dll
-r:C:\Users\buildmaster\.nuget\packages\dotnetzip\1.13.3\lib\net40\DotNetZip.dll
-r:E:\prod\ts-build\BuildDrop\Work\bin_x64\Test\TestAdapters\FarmiCaseHelpers.dll
-r:C:\Users\buildmaster\.nuget\packages\foq\1.7.3\Lib\net45\Foq.dll
-r:C:\Users\buildmaster\.nuget\packages\fsharp.core\4.5.2\lib\net45\FSharp.Core.dll
-r:C:\Users\buildmaster\.nuget\packages\fsharp.core\4.5.2\lib\net45\FSharp.Core.resources.dll
-r:C:\Users\buildmaster\.nuget\packages\fsharp.data.signed\3.1.2\lib\net45\FSharp.Data.dll
-r:C:\Users\buildmaster\.nuget\packages\microsoft.codecoverage\15.9.0\lib\net45\Microsoft.VisualStudio.CodeCoverage.Shim.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\mscorlib.dll
-r:C:\Users\buildmaster\.nuget\packages\newtonsoft.json\11.0.2\lib\net45\Newtonsoft.Json.dll
-r:C:\Users\buildmaster\.nuget\packages\nunit\3.12.0\lib\net45\nunit.framework.dll
-r:C:\Users\buildmaster\.nuget\packages\parsediff\2.0.2\lib\netstandard2.0\ParseDiff.dll
-r:C:\Users\buildmaster\.nuget\packages\protobuf-net\2.4.0\lib\net40\protobuf-net.dll
-r:C:\Users\buildmaster\.nuget\packages\restsharp\106.6.9\lib\net452\RestSharp.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\System.Configuration.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\System.Core.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\System.Data.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\System.dll
-r:C:\Users\buildmaster\.nuget\packages\system.drawing.common\4.5.1\ref\net461\System.Drawing.Common.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\System.Drawing.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\System.IO.Compression.FileSystem.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\System.Management.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\System.Numerics.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\System.Runtime.Serialization.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\System.ServiceModel.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\System.Web.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\System.Xml.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\System.Xml.Linq.dll
-r:E:\prod\ts-build\BuildDrop\Work\bin_x64\Test\TestAdapters\TafSdk.dll
-r:C:\Users\buildmaster\.nuget\packages\teamcityrestapi\2.6.1\lib\netstandard2.0\TeamcityRestApi.dll
-r:C:\Users\buildmaster\.nuget\packages\teamcityresttypes\2.6.0\lib\netstandard2.0\TeamcityRestTypes.dll
-r:C:\Users\buildmaster\.nuget\packages\testimpactapi\5.2.0\lib\netstandard2.0\TestImpactApi.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\Microsoft.Win32.Primitives.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\netstandard.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.AppContext.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Collections.Concurrent.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Collections.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Collections.NonGeneric.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Collections.Specialized.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.ComponentModel.Annotations.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.ComponentModel.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.ComponentModel.EventBasedAsync.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.ComponentModel.Primitives.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.ComponentModel.TypeConverter.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Console.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Data.Common.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Diagnostics.Contracts.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Diagnostics.Debug.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Diagnostics.FileVersionInfo.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Diagnostics.Process.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Diagnostics.StackTrace.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Diagnostics.TextWriterTraceListener.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Diagnostics.Tools.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Diagnostics.TraceSource.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Drawing.Primitives.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Dynamic.Runtime.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Globalization.Calendars.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Globalization.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Globalization.Extensions.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.IO.Compression.ZipFile.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.IO.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.IO.FileSystem.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.IO.FileSystem.DriveInfo.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.IO.FileSystem.Primitives.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.IO.FileSystem.Watcher.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.IO.IsolatedStorage.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.IO.MemoryMappedFiles.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.IO.Pipes.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.IO.UnmanagedMemoryStream.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Linq.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Linq.Expressions.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Linq.Parallel.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Linq.Queryable.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Net.Http.Rtc.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Net.NameResolution.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Net.NetworkInformation.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Net.Ping.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Net.Primitives.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Net.Requests.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Net.Security.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Net.Sockets.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Net.WebHeaderCollection.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Net.WebSockets.Client.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Net.WebSockets.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.ObjectModel.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Reflection.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Reflection.Emit.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Reflection.Emit.ILGeneration.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Reflection.Emit.Lightweight.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Reflection.Extensions.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Reflection.Primitives.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Resources.Reader.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Resources.ResourceManager.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Resources.Writer.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Runtime.CompilerServices.VisualC.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Runtime.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Runtime.Extensions.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Runtime.Handles.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Runtime.InteropServices.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Runtime.InteropServices.RuntimeInformation.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Runtime.InteropServices.WindowsRuntime.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Runtime.Numerics.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Runtime.Serialization.Formatters.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Runtime.Serialization.Json.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Runtime.Serialization.Primitives.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Runtime.Serialization.Xml.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Security.Claims.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Security.Cryptography.Algorithms.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Security.Cryptography.Csp.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Security.Cryptography.Encoding.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Security.Cryptography.Primitives.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Security.Cryptography.X509Certificates.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Security.Principal.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Security.SecureString.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.ServiceModel.Duplex.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.ServiceModel.Http.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.ServiceModel.NetTcp.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.ServiceModel.Primitives.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.ServiceModel.Security.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Text.Encoding.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Text.Encoding.Extensions.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Text.RegularExpressions.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Threading.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Threading.Overlapped.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Threading.Tasks.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Threading.Tasks.Parallel.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Threading.Thread.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Threading.ThreadPool.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Threading.Timer.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.ValueTuple.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Xml.ReaderWriter.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Xml.XDocument.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Xml.XmlDocument.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Xml.XmlSerializer.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Xml.XPath.dll
-r:C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.7.2\Facades\System.Xml.XPath.XDocument.dll
--target:library
--warn:4
--warnaserror
--warnaserror:3239,76
--fullpaths
--flaterrors
--subsystemversion:6.00
--highentropyva+
--targetprofile:mscorlib
--nocopyfsharpcore
--simpleresolution
E:\prod\BuildAgent\temp\buildTmp\.NETFramework,Version=v4.7.2.AssemblyAttributes.fs
E:\prod\ts-build\MSBuild\..\ObjDrop\Work\Debug\x64\FarmiCaseHelpers.Test\FarmiCaseHelpers.Test.AssemblyInfo.fs
ValidateReferenceFilesTest.fs



With this callstack
 	ntdll.dll!773a04ec()	Unknown
 	ntdll.dll![Frames below may be incorrect and/or missing, no symbols loaded for ntdll.dll]	Unknown
 	mscorlib.ni.dll!602388d1()	Unknown
 	mscorlib.ni.dll!602388d1()	Unknown
 	[Managed to Native Transition]	
 	mscorlib.dll!System.IO.FileStream.ReadFileNative(Microsoft.Win32.SafeHandles.SafeFileHandle handle, byte[] bytes, int offset, int count, System.Threading.NativeOverlapped* overlapped, out int hr)	Unknown
 	mscorlib.dll!System.IO.FileStream.ReadCore(byte[] buffer, int offset, int count)	Unknown
 	mscorlib.dll!System.IO.FileStream.Read(byte[] array, int offset, int count)	Unknown
>	FSharp.Compiler.Private.dll!FSharp.Compiler.AbstractIL.Internal.Library.Shim.File.ReadBinaryChunk.Static(string fileName, int start, int len) Line 1374	F#
 	FSharp.Compiler.Private.dll!FSharp.Compiler.AbstractIL.IL.ILResource.GetBytes() Line 2222	F#
 	FSharp.Compiler.Private.dll!FSharp.Compiler.CompileOps.optDataReaders@3711-1.Invoke(Microsoft.FSharp.Core.Unit unitVar0) Line 3711	F#
 	FSharp.Compiler.Private.dll!FSharp.Compiler.CompileOps.optdata@4445.Invoke(Microsoft.FSharp.Core.Unit unitVar) Line 4451	F#
 	mscorlib.dll!System.Lazy<Microsoft.FSharp.Core.FSharpOption<System.Lazy<FSharp.Compiler.Optimizer.ModuleInfo>>>.CreateValue()	Unknown
 	mscorlib.dll!System.Lazy<Microsoft.FSharp.Core.FSharpOption<System.Lazy<FSharp.Compiler.Optimizer.ModuleInfo>>>.LazyInitValue()	Unknown
 	mscorlib.dll!System.Lazy<Microsoft.FSharp.Core.FSharpOption<System.Lazy<FSharp.Compiler.Optimizer.ModuleInfo>>>.Value.get()	Unknown
 	FSharp.Compiler.Private.dll!FSharp.Compiler.CompileOptions.AddExternalCcuToOpimizationEnv(FSharp.Compiler.TcGlobals.TcGlobals tcGlobals, FSharp.Compiler.Optimizer.IncrementalOptimizationEnv optEnv, FSharp.Compiler.CompileOps.ImportedAssembly ccuinfo) Line 1684	F#
 	FSharp.Compiler.Private.dll!FSharp.Compiler.CompileOptions.GetInitialOptimizationEnv@1695.Invoke(FSharp.Compiler.Optimizer.IncrementalOptimizationEnv arg10@, FSharp.Compiler.CompileOps.ImportedAssembly arg20@) Line 1695	F#
 	FSharp.Core.dll!Microsoft.FSharp.Collections.ListModule.Fold<FSharp.Compiler.CompileOps.ImportedAssembly, FSharp.Compiler.Optimizer.IncrementalOptimizationEnv>(Microsoft.FSharp.Core.FSharpFunc<FSharp.Compiler.Optimizer.IncrementalOptimizationEnv, Microsoft.FSharp.Core.FSharpFunc<FSharp.Compiler.CompileOps.ImportedAssembly, FSharp.Compiler.Optimizer.IncrementalOptimizationEnv>> folder, FSharp.Compiler.Optimizer.IncrementalOptimizationEnv state, Microsoft.FSharp.Collections.FSharpList<FSharp.Compiler.CompileOps.ImportedAssembly> list) Line 221	F#
 	FSharp.Compiler.Private.dll!FSharp.Compiler.Driver.main2a<FSharp.Compiler.AbstractIL.Internal.Library.CompilationThreadToken, FSharp.Compiler.TcGlobals.TcGlobals, FSharp.Compiler.TypeChecker.TopAttribs, Microsoft.FSharp.Core.FSharpOption<string>, string, Microsoft.FSharp.Core.FSharpOption<FSharp.Compiler.AbstractIL.IL.ILVersionInfo>, FSharp.Compiler.Driver.StrongNameSigningInfo>(FSharp.Compiler.Driver.Args<System.Tuple<FSharp.Compiler.AbstractIL.Internal.Library.CompilationThreadToken, FSharp.Compiler.CompileOps.TcConfig, FSharp.Compiler.CompileOps.TcImports, FSharp.Compiler.CompileOps.TcImports, FSharp.Compiler.TcGlobals.TcGlobals, FSharp.Compiler.ErrorLogger.ErrorLogger, FSharp.Compiler.Tast.CcuThunk, System.Tuple<string, Microsoft.FSharp.Collections.FSharpList<FSharp.Compiler.Tast.TypedImplFile>, FSharp.Compiler.TypeChecker.TopAttribs, Microsoft.FSharp.Core.FSharpOption<string>, string, Microsoft.FSharp.Core.FSharpOption<FSharp.Compiler.AbstractIL.IL.ILVersionInfo>, FSharp.Compiler.Driver.StrongNameSigningInfo, System.Tuple<FSharp.Compiler.ErrorLogger.Exiter>>>> _arg1) Line 2018	F#
 	FSharp.Compiler.Private.dll!FSharp.Compiler.Driver.typecheckAndCompile(FSharp.Compiler.AbstractIL.Internal.Library.CompilationThreadToken ctok, string[] argv, FSharp.Compiler.ReferenceResolver.Resolver legacyReferenceResolver, bool bannerAlreadyPrinted, FSharp.Compiler.AbstractIL.ILBinaryReader.ReduceMemoryFlag reduceMemoryUsage, FSharp.Compiler.CompileOps.CopyFSharpCoreFlag defaultCopyFSharpCore, FSharp.Compiler.ErrorLogger.Exiter exiter, FSharp.Compiler.Driver.ErrorLoggerProvider loggerProvider, Microsoft.FSharp.Core.FSharpOption<Microsoft.FSharp.Core.FSharpFunc<FSharp.Compiler.CompileOps.TcImports, Microsoft.FSharp.Core.Unit>> tcImportsCapture, Microsoft.FSharp.Core.FSharpOption<Microsoft.FSharp.Core.FSharpFunc<System.Tuple<FSharp.Compiler.TcGlobals.TcGlobals, string, FSharp.Compiler.AbstractIL.IL.ILModuleDef>, Microsoft.FSharp.Core.Unit>> dynamicAssemblyCreator) Line 2186	F#
 	fsc.exe!FSharp.Compiler.CommandLineMain.Driver.main(string[] argv) Line 73	F#
 	fsc.exe!FSharp.Compiler.CommandLineMain.main(string[] argv) Line 84	F#
 	[Native to Managed Transition]	
 	mscoreei.dll!62d9fa84()	Unknown
 	mscoree.dll!62e2e80e()	Unknown
 	mscoree.dll!62e343f8()	Unknown
 	kernel32.dll!76bc0419()	Unknown
 	ntdll.dll!7739662d()	Unknown
 	ntdll.dll!773965fd()	Unknown


![image](https://user-images.githubusercontent.com/1041352/66060061-b7c29e80-e545-11e9-8864-62aa1af53c83.png)

**Comment by jmecosta** [comment]
this one, is a local build... another lock
![image](https://user-images.githubusercontent.com/1041352/66061430-26086080-e548-11e9-80a2-36ad43c16aea.png)

**Comment by cartermp** [comment]
@jmecosta We don't really have a good way to consider custom SDKs like this. Neither the F#, project system, nor .NET SDK team build, test, or deliver this specific SDK.

FWIW you don't need this SDK to accomplish central package versioning. A sample of how to centralize  packages and versions is here: https://github.com/davkean/maket

Does that work?

cc @jeffkl

**Comment by jmecosta** [comment]
@cartermp that was my guess! but i dont think its related with the SDK at all since "The last Visual Studio 16.3.2 hangs by switching solution configuration #7691" is blowing up from everywhere. 

The current lock is coming from F# fsc.exe, and im inclined to blame vs 16.3.2 since everything seems to be locking over there.

**Comment by jeffkl** [comment]
`Microsoft.Build.CentralPackageVersions` is doing the same think that Maket is, which is using MSBuild's `<Item Update="" />` logic.  It does run a target as well to ensure that versions are used correctly but I don't think the SDK would cause a UI hang.

**Comment by cartermp** [comment]
Closing this out as the underlying problem (tracked in another issue) has been resolved.

**Comment by jmecosta** [comment]
Thanks Phillip, i can confirm we havent see any problems recently
